### DISCUSSION TITLE
App api key

### DISCUSSION THREAD

**Discussion by yuhaocooper** [etsy/open-api#877] [Q&A]
Hi, I'm developing an app for Etsy and Shopify integration.

I've created 1 app which I've applied for commercial from the start, this is wrong, and was rightly rejected. Then I've created a 2nd app for personal use as I'm just starting the development, and it is was immediately listed as pending for approval being being rejected a few weeks later.

This part confuses me as I wasn't able to use any API key to test or develop at all (using the /v3/application/openapi-ping endpoint) despite creating it as a personal app, and the documentation stating that we should be able to use the API key immediately if it's for personal use.

Can anyone guide me on how one gets an API key to develop Etsy app, or are all developers just shut out of the Etsy API now? And should we simply not work on Etsy till their API access and management is sorted?

To add more information, my app will require access to edit Etsy's listings and receipt endpoints.

This is frustrating.

**Comment by PMStanley** [comment]
I think your best course of action is to contact the developer team directly as it's unlikely you'll get a response here to an account specific issue.

For what it's worth I was unable to get an API key approved for OAuth access until I included an overview of the app I was planning to build in the submission and stood up a website to accompany it. My requests for a key to get started with the API and follow the tutorials were rejected.

**Reply by yuhaocooper** [reply]
Thanks for sharing your pov. I got a description of my app and a website up, but it still got rejected. And I guess a direct contact to the developer team will be best to resolve this.

But it's a frustrating process to gain basic API access, which is hardly ever an issue anywhere else, and I feel this is an unfriendly stance towards 3rd party developers from Etsy that is deterring me from building an app that is dependent on it.
